lib/reqTemplate.js

```diff
@@ -121,6 +151,40 @@ function _createTemplateResolvers(origSpec, subspec, reqPart, path) {
 }
 
 /**
+ * Creates a template resolver functuons for URI part of the spec
+ * @param {object} spec a root request spec object
+ * @returns {[Function]} array of template resolvers which should be applied to resolve URI
+ */
+function createURIResolver(spec) {
+    var setter = _setAtPath('uri', spec);
+    if (/^\{[^\{}]+}$/.test(spec.uri)) {
+        var setValue = function(newReq, value) {
+            if (value.constructor !== URI) {
+                value = new URI(value, {}, false);
+            }
+            setter(newReq, value);
+        };
+        return compileTemplate(spec.uri, setValue, 'params');
+    } else if (/^(?:https?:\/\/)?\{[^\/]+}\//.test(spec.uri)) {
+        // The host is templated - replace it with TAssembly and use URI.expand for path templates
+        var hostTemplate = /^((?:https?:\/\/)?\{[^\/]+}\/)/.exec(spec.uri)[1];
+        return compileTemplate(hostTemplate, setter, 'params').map(function(resolver) {
+            return function(newReq, context) {
+                var uriSpec;
+                resolver(newReq, context);
+                uriSpec = newReq.uri + spec.uri.substr(hostTemplate.length);
```

Comment:
This looks rather expensive, with a new URI template being created for each expansion. Maybe you could reuse the path template somewhat like this, by updating `.urlObj`:

```function
var pathTemplate = new URI(path, {}, true);
return function(newReq, context) {
  var newReq = resolver(newReq, context);
  var newUri = pathTemplate.expand(context.request.params);
  newUri.urlObj = url.parse(newPrefix.uri);
  setter(..)
}
```